pkg/apis/mysql/v1alpha1/types.go

```diff
@@ -27,6 +28,11 @@ const MinimumMySQLVersion = "8.0.11"
 type ClusterSpec struct {
 	// Version defines the MySQL Docker image version.
 	Version string `json:"version"`
+	// MySQLServerImage defines the image to be pulled for the mysqlServer.
+	MySQLServerImage string `json:"mysqlServer"`
+	// ImagePullSecret defines the name of the secret that contains the
+	// required credentials for pulling the MySQLServerImage.
+	ImagePullSecret *v1.LocalObjectReference `json:"imagePullSecret"`
```

Comment:
I suggest we implement as `[]v1.LocalObjectReference` as in the core.

From `v1.PodSpec`:
```go
// ImagePullSecrets is an optional list of references to secrets in the same namespace to use for pulling any of the images used by this PodSpec.
// If specified, these secrets will be passed to individual puller implementations for them to use. For example,
// in the case of docker, only DockerConfig type secrets are honored.
// More info: https://kubernetes.io/docs/concepts/containers/images#specifying-imagepullsecrets-on-a-pod
// +optional
// +patchMergeKey=name
// +patchStrategy=merge
ImagePullSecrets []LocalObjectReference `json:"imagePullSecrets,omitempty" patchStrategy:"merge" patchMergeKey:"name" protobuf:"bytes,15,rep,name=imagePullSecrets"`
```